### PR TITLE
Image Classification Refactoring

### PR DESCRIPTION
**Pull request by akshaykvnit**
Started refactoring image classification code. Not yet done, but @SharathRaparthy and @navidpanchi, take a look at the organization of the code, and let me know if any files need to moved somewhere else. 
Meanwhile, I'll keep changing the code internally without changing the overall directory structure.

### CODE REVIEW

File: image_classification/experiments/no_teacher.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from image_classification.models.custom_resnet import *
+from trainer import *
+
+
+args = get_args(description='No Teacher', mode='train')
+
+torch.manual_seed(args.seed)
+if args.gpu != 'cpu':
+    torch.cuda.set_device(args.gpu)
+    torch.cuda.manual_seed(args.seed)
```
Comment (navidpanchi) [inline image_classification/experiments/no_teacher.py:20]: first set_device and then use manual seed outside the if.

File: image_classification/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import os
+import numpy as np
+import torch
+from fastai.vision import *
+from image_classification.models.custom_resnet import *
```
Comment (navidpanchi) [inline image_classification/utils/utils.py:5]: import as `from image_classification.models import custom_resnet` - why? see the comment below

File: image_classification/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    elif model_name == 'resnet14' : 
+        net = resnet14(pretrained=False, progress=False)
+    elif model_name == 'resnet18' :
+        net = resnet18(pretrained=False, progress=False)
+    elif model_name == 'resnet20' :
+        net = resnet20(pretrained=False, progress=False)
+    elif model_name == 'resnet26' :
+        net = resnet26(pretrained=False, progress=False)
+    elif model_name == 'resnet34' :
+        net = resnet34(pretrained=False, progress=False)
```
Comment (navidpanchi) [inline image_classification/utils/utils.py:93]: since all of them have the same parameters you should use `getattr` function.
``` 
net = getattr(model_name)(pretrained=False, progress=False)
```

File: image_classification/utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    if teach:
+        return teacher, net
+    else:
+        return net
+
+
+def get_accuracy(dataloader, Net):
```
Comment (navidpanchi) [inline image_classification/utils/utils.py:101]: don't use first letter as capital for variables - not a proper style so it can confuse

File: image_classification/experiments/no_teacher.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (akshaykvnit) [reply image_classification/experiments/no_teacher.py:20]: what's the difference? didn't get your point.

File: image_classification/experiments/no_teacher.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                                                                loss_function2=None,
+                                                                optimizer=optimizer,
+                                                                hyper_params=hyper_params,
+                                                                epoch=epoch,
+                                                                savename=savename,
+                                                                best_val_acc=best_val_acc
+                                                                )
+    experiment.log_metric("train_loss", train_loss)
+    experiment.log_metric("val_loss", val_loss)
+    experiment.log_metric("val_acc", val_acc * 100)
```
[the remark this answers is not in the document]
Comment (akshaykvnit) [reply image_classification/experiments/no_teacher.py:67]: I feel like this showcases our training approach more clearly rather than hiding it in another file. Let's see what @SharathRaparthy thinks.

File: image_classification/utils/utils.py (hunk 3 above)
Comment (akshaykvnit) [reply image_classification/utils/utils.py:93]: yes, I was looking at this. Wasn't sure how it worked. I'll try it.

File: image_classification/experiments/no_teacher.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (navidpanchi) [reply image_classification/experiments/no_teacher.py:20]: Yeah no this is correct

File: image_classification/utils/utils.py (hunk 3 above)
Comment (navidpanchi) [reply image_classification/utils/utils.py:93]: It should be:
getattr(model_name, custom_resnet)

Comment (akshaykvnit) [reply image_classification/utils/utils.py]: done. It's actually `getattr(custom_resnet, model_name)`

File: image_classification/experiments/no_teacher.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (SharathRaparthy) [reply image_classification/experiments/no_teacher.py]: Hey! Just maintain the uniformity for all. I agree with @akshaykvnit

**Comment by navidpanchi** [comment]
Is this done?

**Comment by akshaykvnit** [comment]
@navidpanchi You can review the changes. Other than that, @SharathRaparthy is running some verification experiments to check if code works like earlier. So, we'll have to wait for that before merging.
